src/Libraries/RevitNodes/Elements/AdaptiveComponent.cs

```diff
@@ -412,11 +184,28 @@ public static AdaptiveComponent ByParametersOnFace(Autodesk.DesignScript.Geometr
                 throw new ArgumentNullException("familyType");
             }
 
-            return new AdaptiveComponent(uvs.Select(x => new[] { x.U, x.V }).ToArray(), ElementFaceReference.TryGetFaceReference(surface), familyType);
+            var result = GeometryObjectSelector.ByReferenceStableRepresentation(
+                        faceReference.InternalReference.ConvertToStableRepresentation(Document));
+            var surface = result as Surface;
+            if (surface == null)
+            {
+                var surfaces = result as List<Surface>;
+                if (surfaces != null && surfaces.Count() > 0)
+                {
+                    surface = surfaces[0];
```

Comment:
Good question! This is implemented as such because ByReferenceStableRepresentation is returning an object which can be a single element or a list. Right now most probably it is a surface list which contains only one surface.